mapping_parenting_tech/analysis/prototyping/play_store/get_play_store_apps.py

```diff
@@ -7,141 +7,670 @@
+    return list(set(app_id_list))
 
-# %% [markdown]
-# Retrieve app details
 
 # %%
-all_app_details = dict()
-remove_apps = list()
-review_counter = 0
-most_reviews = {"id": 0, "count": 0}
+def app_snowball(seed_app_id: str, depth: int = 5, __current_depth: int = 1) -> list:
+    """
+    Retrieves ids of Play Store apps related to `seed_app_id` by calling itself recursively.
 
-for app_id in tqdm(output_ids["Parenting apps"], desc="Retrieving app details"):
-    try:
-        app_details = app(
-            app_id, lang="en", country="gb"  # defaults to 'en'  # defaults to 'us'
-        )
-        all_app_details.update({app_id: app_details})
-        review_counter += app_details["reviews"]
-        if app_details["reviews"] > most_reviews["count"]:
-            most_reviews["id"] = app_id
-            most_reviews["count"] = app_details["reviews"]
+    Args:
+        seed_app_id: str - the app id of the app of interest
+        depth: int, default = 5 - the depth of recursion. This will increase the number of apps interrogated (and
+        therefore the time taken for the initial call to complete) exponentially
+        current_depth: used for recursion, should be left blank by user
 
-    except Exception as e:  # needs modifying to capture specific errors
-        logging.info(f"Error on app id {app_id}: {e} {repr(e)}")
-        remove_apps.append(app_id)
+    Returns:
+        a list of app ids
+    """
 
-for app_id in remove_apps:
-    output_ids["Parenting apps"].remove(app_id)
+    app_details = app(seed_app_id, country="gb")
+    similar_apps = app_details["similarApps"]
+
+    snowball = set()
+    snowball.update(similar_apps)
+
+    if __current_depth < depth:
+        for this_app in similar_apps:
+            snowball.update(app_snowball(this_app, depth, (__current_depth + 1)))
+
+    return list(snowball)
+
+
+# %%
+def save_app_ids(
+    app_list: list, app_category: str, filename: str, output_path: Path = APP_IDS_PATH
+) -> bool:
+    """
+    Saves a list of app ids to a given CSV file. If the file already exists, it is overwritten.
+
+    Args:
+        app_list: list - a list of app ids to be saved
+        app_category: str - the title of the app category being saved, used as the header at the top of the list
+        filename: str - the filename to be used, including extension
+        output_path: Path - destination folder; default is `outputs/data`
+
+    Returns:
+        True if executed successfully
+
+    """
+
+    output_target = output_path / filename
+
+    with open(output_target, "wt", newline="\n") as output_file:
+        csv_writer = csv.writer(output_file)
+        csv_writer.writerow([app_category])
+        # NB can't use writerows() as it will treat each string as a list and split it into comma-separated letters
+        # hence why `app_id` (and `app_category`, above) is encapsulated in square brackets
+        for app_id in app_list:
+            csv_writer.writerow([app_id])
+
```

Comment:
On a second thought, just returning True is a bit uninformative.
You could add a logging output as well before terminating the function, eg
```suggestion

    logging.info(f"App ids saved in {output_target}")
```